src/imageio/format/jxl.c

```diff
@@ -0,0 +1,749 @@
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_SRGB;
+      }
+      else if(over_type == DT_COLORSPACE_INFRARED)
+      {
+        primaries = sRGB_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_LINEAR;
+      }
+      else if(over_type == DT_COLORSPACE_DISPLAY)
+      {
+        primaries = sRGB_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_SRGB;
+      }
+      else if(over_type == DT_COLORSPACE_DISPLAY2)
+      {
+        primaries = sRGB_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_SRGB;
+      }
+      else if(over_type == DT_COLORSPACE_LIN_REC2020)
+      {
+        primaries = Rec2020_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_LINEAR;
+      }
+      else if(over_type == DT_COLORSPACE_PQ_REC2020)
+      {
+        primaries = Rec2020_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_PQ;
+      }
+      else if(over_type == DT_COLORSPACE_HLG_REC2020)
+      {
+        primaries = Rec2020_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_HLG;
+      }
+      else if(over_type == DT_COLORSPACE_REC709)
+      {
+        primaries = Rec709_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_709;
+      }
+      else if(over_type == DT_COLORSPACE_LIN_REC709)
+      {
+        primaries = Rec709_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_LINEAR;
+      }
+      else if(over_type == DT_COLORSPACE_ADOBERGB)
+      {
+        primaries = Adobe_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_GAMMA;
+        color_encoding.gamma = 2.19921875;
+      }
+      else if(over_type == DT_COLORSPACE_PQ_P3)
+      {
+        primaries = P3_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_PQ;
+      }
+      else if(over_type == DT_COLORSPACE_HLG_P3)
+      {
+        primaries = P3_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_HLG;
+      }
+      else if(over_type == DT_COLORSPACE_PROPHOTO_RGB)
+      {
+        primaries = ProPhoto_Primaries;
+        color_encoding.transfer_function = JXL_TRANSFER_FUNCTION_LINEAR;
+      }
+      else
+      {
+        write_icc = true;
+      }
+
+      if(!write_icc)
+      {
+        color_encoding.primaries = JXL_PRIMARIES_CUSTOM;
+        color_encoding.primaries_red_xy[0] = primaries.Red.x;
+        color_encoding.primaries_red_xy[1] = primaries.Red.y;
+        color_encoding.primaries_green_xy[0] = primaries.Green.x;
+        color_encoding.primaries_green_xy[1] = primaries.Green.y;
+        color_encoding.primaries_blue_xy[0] = primaries.Blue.x;
+        color_encoding.primaries_blue_xy[1] = primaries.Blue.y;
+        // TODO: safe to simply put pipe->icc_intent here?
+        color_encoding.rendering_intent = JXL_RENDERING_INTENT_PERCEPTUAL;
```

Comment:
For the same reasons as above, the standard output color spaces should use existing CICP/H.273 enums like JXL_PRIMARIES_SRGB etc. Please leave the custom values just for ones that cannot be covered.

Also, please move this section to be exclusive w/ the `if(write_icc)` block below as it expresses the intent better.